bigquery/google/cloud/bigquery/table.py

```diff
@@ -759,3 +760,61 @@ def __init__(self, resource):
         # time is in milliseconds since the epoch.
         self.oldest_entry_time = _datetime_from_microseconds(
             1000.0 * int(resource['oldestEntryTime']))
+
+
+class Row(object):
+    """A BigQuery row.
+
+    Values can be accessed by position (index), by key like a dict,
+    or as properties.
+
+    :type values: tuple
+    :param values:  the row values
+
+    :type field_to_index: dict
+    :param field_to_index:  a mapping from schema field names to indexes
+    """
+
+    # Choose unusual field names to try to avoid conflict with schema fields.
+    __slots__ = ('_xxx_values', '_xxx_field_to_index')
+
+    def __init__(self, values, field_to_index):
+        self._xxx_values = values
+        self._xxx_field_to_index = field_to_index
+
+    def values(self):
+        return self._xxx_values
```

Comment:
This means the caller can mutate your **internal** representation if it is a mutable type. Do you want that? (I suppose that's the users' "fault", since the docstring says `values` must be a `tuple`.

Reply:
Talked to @jonparrott about this. I think we can let the user accept this risk if they use the class improperly.